src/test/java/org/springframework/integration/kafka/support/KafkaConsumerContextIntegrationTests.java

```diff
@@ -0,0 +1,152 @@
+import java.util.Map;
+import java.util.Properties;
+import java.util.UUID;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.springframework.integration.kafka.outbound.KafkaProducerMessageHandler;
+import org.springframework.integration.kafka.rule.KafkaEmbedded;
+import org.springframework.integration.kafka.rule.KafkaRule;
+import org.springframework.integration.kafka.serializer.common.StringDecoder;
+import org.springframework.integration.kafka.serializer.common.StringEncoder;
+import org.springframework.integration.kafka.util.EncoderAdaptingSerializer;
+import org.springframework.integration.kafka.util.TopicUtils;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+
+import kafka.admin.AdminUtils;
+import kafka.common.TopicExistsException;
+import kafka.consumer.ConsumerConfig;
+import kafka.serializer.Encoder;
+
+/**
+ * @author Gary Russell
+ * @since 1.3
+ */
+public class KafkaConsumerContextIntegrationTests {
+
+	private static final String TOPIC = "springintegrationtestInbound";
+
+	@Rule
+	public KafkaRule kafkaRule = new KafkaEmbedded(1);
+
+	@After
+	public void tearDown() {
+		try {
+			AdminUtils.deleteTopic(kafkaRule.getZkClient(), TOPIC);
+		}
+		catch (Exception e) {
+		}
+	}
+
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	@Test
+	public void test() throws Exception {
+		// create the topic
+
+		try {
+			TopicUtils.ensureTopicCreated(kafkaRule.getZookeeperConnectionString(), TOPIC, 1, 1);
+		}
+		catch (TopicExistsException e) {
+			// do nothing
+		}
+
+		final String suffix = UUID.randomUUID().toString();
+
+		KafkaProducerContext producerContext = createProducerContext();
+		KafkaProducerMessageHandler handler =
+				new KafkaProducerMessageHandler(producerContext);
+
+		Message<String> msg = MessageBuilder.withPayload("foo" + suffix)
+				.setHeader(KafkaHeaders.MESSAGE_KEY, "3")
+				.setHeader(KafkaHeaders.TOPIC, TOPIC)
+				.build();
+		handler.handleMessage(msg);
+
+		final KafkaConsumerContext kafkaConsumerContext = createConsumerContext();
+
+		Message received = kafkaConsumerContext.receive();
+		assertNotNull(received);
+		Map<?,?> payload = (Map<?,?> )received.getPayload();
+		assertNotNull(payload);
+		Map<?,?> topicMap = (Map<?,?>) payload.get(TOPIC);
+		assertNotNull(topicMap);
+		List<?> list = (List<?>) topicMap.get(0);
+		assertNotNull(list);
+		assertThat(list.size(), greaterThanOrEqualTo(1));
+		assertEquals("foo" + suffix, list.get(0));
+	}
+
+	private KafkaConsumerContext createConsumerContext() {
+		final KafkaConsumerContext kafkaConsumerContext = new KafkaConsumerContext();
+
+		final Map<String, ConsumerConfiguration> map = new HashMap<String, ConsumerConfiguration>();
+		ConsumerMetadata consumerMetadata = new ConsumerMetadata<>();
+		consumerMetadata.setGroupId("foo");
+		consumerMetadata.setValueDecoder(new StringDecoder());
+		consumerMetadata.setKeyDecoder(new StringDecoder());
```

Comment:
I mean that `ProducerMetadata` is based on the `Serializer` since upgrade to Kafka `0.8.2`, but this `ConsumerMetadata` is still on the `Decoder`.
Doesn't look symmetrical...